src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -144,15 +168,16 @@ object DataTypes {
       val nonExistingFields = findNonExistingFields(query, entity, tezosPlatformDiscovery)
       val invalidTypeAggregationField = findInvalidAggregationTypeFields(query, entity, tezosPlatformDiscovery)
 
-      for {
+      val result = for {
         invalidNonExistingFields <- nonExistingFields
         invalidAggregationFieldForTypes <- invalidTypeAggregationField
       } yield {
         invalidNonExistingFields ::: invalidAggregationFieldForTypes match {
-          case Nil => Right(query)
-          case wrongFields => Left(wrongFields)
+          case Nil => Right(replaceTimestampInPredicates(entity, query, tezosPlatformDiscovery))
```

Comment:
Suggestion: replace timestamps directly where `query` is defined, and make the val `lazy`

Reply:
else you can simply put the `replaceTimestamp...` in the for-comp, to remove the extra layer to peel out with `bisequence` down here

Reply:
you are right, I was overthinking it so to be safe I wanted to make `replaceTimestampInPredicates` a last call so if there were validation errors this method would not have a chance to fail, but it still wouldn't